piqe_ocp_lib/api/resources/ocp_health_checker.py

```diff
@@ -0,0 +1,318 @@
+        for deployment in deployment_response.items:
+            if deployment.metadata.name == "router-default":
+                replicas = deployment.status.replicas
+                available_replicas = deployment.status.availableReplicas
+                ready_replicas = deployment.status.readyReplicas
+        if replicas == available_replicas == ready_replicas:
+            is_replicas_count_matching = True
+        unhealthy_router_info.update({"router_replicas": is_replicas_count_matching})
+        logger.info("Is router deployment replicas count matching : %s", is_replicas_count_matching)
+
+        logger.info("Check overall health of router operator")
+        if is_router_pod_healthy and is_replicas_count_matching:
+            is_router_healthy = True
+
+        return is_router_healthy, unhealthy_router_info
+
+    def check_image_registry_health(self):
+        """
+        Check openshift cluster image registry
+        - Check if image registry pod is running fine in openshift-image-registry namespace
+        - Check if deployments of image registry has matching number of replicas
+        :return:
+        """
+        logger.info("Check health of openshift image registry pods")
+        is_image_registry_healthy = False
+        unhealthy_image_registry_info = dict()
+        is_image_registry_pods_healthy = False
+        unhealthy_pods = dict()
+        pods_response = self.ocp_pod.list_pods_in_a_namespace(namespace="openshift-image-registry")
+        for pod in pods_response.items:
+            if "cluster-image-registry-operator" in pod.metadata.name or \
+                    "image-registry" in pod.metadata.name:
+                for condition in pod["status"]["conditions"]:
+                    if condition["type"] == "Ready" and condition["status"] == "False":
+                        unhealthy_pods[pod["metadata"]["name"]] = condition["status"]
+
+        if len(unhealthy_pods.values()) == 0:
+            is_image_registry_pods_healthy = True
+        unhealthy_image_registry_info.update({"image_registry_pod": unhealthy_pods.keys()})
+        logger.info("Is image registry pod/s healthy : %s", is_image_registry_pods_healthy)
+
+        logger.info("Check replicas count of openshift image registry deployment are matching")
+        is_replicas_count_matching = False
+        replica_count_dict = dict()
+        deployment_response = \
+            self.ocp_deployment.list_all_deployments_in_a_namespace(namespace="openshift-image-registry")
+        for deployment in deployment_response.items:
+            if deployment.metadata.name == "cluster-image-registry-operator" or \
+                    deployment.metadata.name == "image-registry":
+                replicas = deployment.status.replicas
+                available_replicas = deployment.status.availableReplicas
+                ready_replicas = deployment.status.readyReplicas
```

Comment:
There may be a refactoring opportunity between this and the code on line 108.

Reply:
Same as above